File: src/test/scala/com/gitlab/dhorman/cryptotrader/util/BigDecimalUtil.scala
package com.gitlab.dhorman.cryptotrader.util
import scala.math.BigDecimal.RoundingMode

object BigDecimalUtil {
  implicit final class BigDecimalUtil(val num: BigDecimal) extends AnyVal {
    def cut3: BigDecimal = num.setScale(3, RoundingMode.HALF_DOWN)
    def cut8: BigDecimal = num.setScale(8, RoundingMode.HALF_DOWN)
  }
}

File: src/test/kotlin/com/gitlab/dhorman/cryptotrader/util/BigDecimalUtil.kt
package com.gitlab.dhorman.cryptotrader.util

import java.math.BigDecimal
import java.math.RoundingMode

val BigDecimal.cut3 get() = this.setScale(3, RoundingMode.HALF_DOWN)
val BigDecimal.cut8 get() = this.setScale(8, RoundingMode.HALF_DOWN)